### PR TITLE
Dm/drop usage sq api before 56

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

**Comment by henryju** [comment]
I'm not sure I'm the best to say if changes are ok, but LGTM

File: src/main/java/org/sonar/squidbridge/SquidAstVisitorContext.java [hunk 1]
```diff
@@ -50,7 +47,7 @@
    * @param message           message describing the violation, can be formatted (see java.text.MessageFormat)
    * @param messageParameters optional message parameters (see java.text.MessageFormat)
    */
-  public abstract void createFileViolation(CodeCheck check, String message, Object... messageParameters);
+  public abstract void createFileViolation(Object check, String message, Object... messageParameters);
```
Comment (nixel2007) [inline src/main/java/org/sonar/squidbridge/SquidAstVisitorContext.java]: Why don't migrate to SquidCheck<Grammar> or SquidAstVisitor<Grammar> ?

File: src/main/java/org/sonar/squidbridge/annotations/AnnotationBasedRulesDefinition.java [hunk 2]
```diff
@@ -106,10 +107,6 @@ public void addRuleClasses(boolean failIfSqaleNotFound, boolean failIfNoExplicit
     setupExternalNames(newRules);
   }
 
-  private boolean isSqaleAnnotated(Class<?> ruleClass) {
-    return getSqaleSubCharAnnotation(ruleClass) != null || getNoSqaleAnnotation(ruleClass) != null;
-  }
-
   @VisibleForTesting
   NewRule newRule(Class<?> ruleClass, boolean failIfNoExplicitKey) {
     org.sonar.check.Rule ruleAnnotation = AnnotationUtils.getAnnotation(ruleClass, org.sonar.check.Rule.class);
```
Comment (SonarTech) [inline src/main/java/org/sonar/squidbridge/annotations/AnnotationBasedRulesDefinition.java]: MINOR Replace this fully qualified name with "Rule" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942)
MINOR Replace this fully qualified name with "Rule" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1942)

File: src/main/java/org/sonar/squidbridge/api/CheckMessage.java [hunk 3]
```diff
@@ -103,7 +83,6 @@ public String getDefaultMessage() {
     return messageArguments;
   }
 
-  @Override
   public String getText(Locale locale) {
```
Comment (SonarTech) [inline src/main/java/org/sonar/squidbridge/api/CheckMessage.java]: MAJOR Remove this unused method parameter "locale". [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)

**Comment by SonarTech** [comment]
SonarQube analysis reported 51 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 49 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractXPathCheck.java#L52](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/checks/AbstractXPathCheck.java#L52): Remove this expression which always evaluates to "true" [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2589)
1. ![MINOR][MINOR] [BranchCoverageCheck.java#L50](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/BranchCoverageCheck.java#L50): Remove this use of "Resource"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BranchCoverageCheck.java#L50](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/BranchCoverageCheck.java#L50): Remove this use of "DecoratorContext"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BranchCoverageCheck.java#L50](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/BranchCoverageCheck.java#L50): Remove this use of "ResourcePerspectives"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BranchCoverageCheck.java#L51](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/BranchCoverageCheck.java#L51): Remove this use of "MeasureUtils"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BranchCoverageCheck.java#L53](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/BranchCoverageCheck.java#L53): Remove this use of "MeasureUtils"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BranchCoverageCheck.java#L54](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/BranchCoverageCheck.java#L54): Remove this use of "MeasureUtils"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BranchCoverageCheck.java#L62](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/BranchCoverageCheck.java#L62): Remove this use of "Resource"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [BranchCoverageCheck.java#L62](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/BranchCoverageCheck.java#L62): Remove this use of "ResourcePerspectives"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [CommentDensityCheck.java#L49](https://github.com/SonarSource/sslr-squid-bridge/blob/684fb610b1f5b53eac7e0b4e1169eecb6de54e1f/src/main/java/org/sonar/squidbridge/commonrules/internal/checks/CommentDensityCheck.java#L49): Remove this use of "Resource"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by m-g-sonar** [comment]
closed as covered by #15 .

**Pull request closed without merging**